### ISSUE TITLE
HTTP API for inspect

### ISSUE THREAD

**Issue by sirex** [atviriduomenys/spinta#477]
Create `POST /:inspect` API endpoint, which accepts a `multipart/form-data` form and returns manifest in CSV/HTML or other specified format. Return format can be specified via `?format()` or via `Accept` header.

Form should accept following fields:

- `manifest` - a manifest type, like `csv`, `json`, ... (see [list of available formats](https://github.com/atviriduomenys/katalogas/issues/345)). If not given, use `tabular`.
- `dataset` - dataset name, that will be used in resulting manifest file, if not given, use `dataset1`.
- `url` or `file` with a schema, to read data structure from, one or the other must be provided.

If there are errors, provide errors in JSON or HTML format as specified in `?format()` or `Accept:`.

### Epic

- https://github.com/atviriduomenys/spinta/issues/247

### Related

- https://github.com/atviriduomenys/katalogas/issues/345

**Comment by sirex** [comment]
Probably `multipart/form-data` won't work, so better way to do same request as described in https://github.com/atviriduomenys/spinta/issues/475.

**Comment by sirex** [comment]
Ignore previouse comment, because, it looks, that `multipart/form-data` is the better way to implement this, because we need to pass multiple parameters in order to make this work.

Form should accept following fields:

- `dataset` - required, a dataset name
- `manifest.type` - required if `resource.type` is not given, a manifest type, like `csv`, `json`, ... (see [list of available formats](https://github.com/atviriduomenys/katalogas/issues/345)). If not given, use `csv` tabular format.
- `manifest.file` - required if `manifest.type` is given, a manifest file in specified `manifest.type` format, this will be used as a main manifest, that can be updated from a resource.
- `manifest.source` - required if `manifest.file` is not given, a url containing manifest file
- `resource.type` - required if `manifest.type` is not given, another manifest, that supports inspect from a resource, same as `manifest.type`.
- `resource.file` - required if `resource.source` is not given, a manifest file of `resource.type` type
- `resource.source` - required if `resource.file` is not given, a url containing manifest file
- `resource.prepare` - optional formula to prepare resource before reading


Basically we have this:

    spinta inspect {manifest.file|manifest.source} -r {resource.type} {resource.file|manirfest.source} -f {resource.prepare}

In addition we give `dataset` to put into manifest file instead of a generated dataset name.

One of `manifest.*` or `resource.*` is required, but also both can be given. If boths ar given, then `manifest` is updated from `resource`.

If only `manifest` is given, then we read resources if any given from `manifest` table.
